lib/components/map/connected-transitive-overlay.js

```diff
@@ -1,44 +1,43 @@
-import coreUtils from '@opentripplanner/core-utils'
-import TransitiveCanvasOverlay from '@opentripplanner/transitive-overlay'
+import TransitiveOverlay from '@opentripplanner/transitive-overlay'
+import React, { Component } from 'react'
 import { connect } from 'react-redux'
 
-import { getActiveSearch, getActiveItinerary, getActiveItineraries } from '../../util/state'
+import { getActiveSearch, getTransitiveData } from '../../util/state'
+
+/**
+ * Wrapper for TransitiveOverlay that avoids rerenders by checking transitive
+ * data computed from redux store.
+ */
+class TransitiveCanvasOverlay extends Component {
+  shouldComponentUpdate (nextProps) {
+    return nextProps.transitiveData !== this.props.transitiveData
+  }
```

Comment:
Is this still needed?

Reply:
Maybe, although it is not as beneficial as previous changes. An instance of where this may be useful is for [this query](http://localhost:9966/#/?ui_activeSearch=izrz4hwb7&ui_activeItinerary=3&fromPlace=117%20East%20Amelia%20Street%2C%20Orlando%2C%20FL%2C%20USA%3A%3A28.54930782241189%2C-81.3760645495495&toPlace=2915%20Flowertree%20Road%2C%20Orlando%2C%20FL%2C%20USA%3A%3A28.452280278349317%2C-81.3439156394452&date=2021-06-04&time=14%3A46&arriveBy=false&mode=WALK%2CBUS%2CRAIL&showIntermediateStops=true&maxWalkDistance=3219&optimize=TRANSFERS&walkSpeed=1.34&ignoreRealtimeUpdates=true&companies=&bannedRoutes=&numItineraries=3&otherThanPreferredRoutesPenalty=900&preferredRoutes=) where the first two "Walk to rail" itineraries appear identical but are re-rendered when you hover from one to the other.

Reply:
I removed `shouldComponentUpdate` in https://github.com/opentripplanner/otp-react-redux/pull/366/commits/61236f217c33cd0450f64b3210af75f1c3d71c59, noting the enhancement to the re-rendering process in #381 instead.

Reply:
Oh wow, TIL that [redux already does this shouldComponentUpdate check for connected components](https://redux.js.org/faq/react-redux#why-isnt-my-component-re-rendering-or-my-mapstatetoprops-running), so yeah this should be removed.